src/TestFramework/TestFramework/Logger.cs

```diff
@@ -36,13 +36,34 @@ public static void LogMessage(string format, params object?[] args)
         Guard.NotNull(format);
         Guard.NotNull(args);
 
+        Delegate[] delegates = OnLogMessage.GetInvocationList();
+
+        if (delegates.Length == 0)
```

Comment:
Can this ever happen? I'd assume if there is nothing to invoke, we will find OnLogMessage null?

Reply:
good point. i will check

Reply:
u r correct. i will fix